### PR TITLE
Handle deprecated rules (closes #112)

### LINKED ISSUE #112: Handle deprecated rules
- Extending `prettier`:
  - `no-spaced-func` deprecated in `eslint` since version `3.3.0`
    - https://eslint.org/docs/rules/no-spaced-func
  - `indent-legacy` is deprecated in `eslint` since version `4.0.0`
    - https://github.com/eslint/eslint/pull/8286
- Extending `prettier/react`:
  - `react/jsx-space-before-closing` deprecated in `eslint-plugin-react` since version `7.0.0`:  
    - [eslint-plugin-react changelog](https://github.com/yannickcr/eslint-plugin-react/blob/master/CHANGELOG.md#700---2017-05-06)

Not sure what's the policy here for deprecated rules.  
My use case is to catch deprecated rules in an existing config using [eslint-find-rules](https://github.com/sarbbottam/eslint-find-rules).  
The issue is that projects have configurations with deprecated rules and ESLint does not warn about it.

So I'd like to catch such cases, possible with:

```
npx eslint-find-rules -d .eslintrc
```

Ideally this would exit with a `0` value and when I bump ESLint or related configs/plugins, it'll exit with non `0` and show me which rules are deprecated.

Right now it's not possible when extending `eslint-config-prettier`:

.eslintrc:
```
{
  "extends": ["prettier"]
}
```

```
$ npx eslint-find-rules -d .eslintrc

deprecated rules

indent-legacy    no-spaced-func
```

### PR DESCRIPTION
**Pull request by alexilyaev**
- Conditionally add deprecated rules based on environment variable

Can now pass an environment variable to tell `eslint-config-prettier` not to add deprecated rules.

```shell
ESLINT_CONFIG_PRETTIER_NO_DEPRECATED=1 npx eslint-find-rules -d .eslintrc
```

## Details

- See issue #112 
- Reasoning and alternatives: https://github.com/prettier/eslint-config-prettier/pull/120#issuecomment-544188602

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,93 +1,103 @@
...
+      "switch-colon-spacing": "off",
+      "template-curly-spacing": "off",
+      "template-tag-spacing": "off",
+      "unicode-bom": "off",
+      "wrap-iife": "off",
+      "wrap-regex": "off",
+      "yield-star-spacing": "off"
+    },
+    !noDeprecated && {
+      // Was deprecated in eslint since version 4.0.0
```
Comment (lydell) [inline index.js]:
```suggestion
      // Deprecated since version 4.0.0.
```

File: index.js [hunk 2]
```diff
@@ -1,93 +1,103 @@
...
+      // https://github.com/eslint/eslint/pull/8286
+      "indent-legacy": "off",
+      // Was deprecated in eslint since version 3.3.0
```
Comment (lydell) [inline index.js]:
```suggestion
      // Deprecated in eslint since version 3.3.0.
```

File: react.js [hunk 3]
```diff
@@ -1,21 +1,29 @@
...
+      "react/jsx-indent": "off",
+      "react/jsx-indent-props": "off",
+      "react/jsx-max-props-per-line": "off",
+      "react/jsx-one-expression-per-line": "off",
+      "react/jsx-props-no-multi-spaces": "off",
+      "react/jsx-tag-spacing": "off",
+      "react/jsx-wrap-multilines": "off"
+    },
+    !noDeprecated && {
+      // Was deprecated in eslint-plugin-react since version 7.0.0
```
Comment (lydell) [inline react.js]:
```suggestion
      // Deprecated since version 7.0.0.
```

File: index.js [hunk 4]
```diff
@@ -1,93 +1,103 @@
 "use strict";
 
+const noDeprecated = process.env.ESLINT_CONFIG_PRETTIER_NO_DEPRECATED;
```
Comment (lydell) [inline index.js]:
```suggestion
const includeDeprecated = !process.env.ESLINT_CONFIG_PRETTIER_NO_DEPRECATED;
```

File: index.js [hunk 5]
```diff
@@ -1,93 +1,103 @@
...
+      "space-unary-word-ops": "off",
+      "switch-colon-spacing": "off",
+      "template-curly-spacing": "off",
+      "template-tag-spacing": "off",
+      "unicode-bom": "off",
+      "wrap-iife": "off",
+      "wrap-regex": "off",
+      "yield-star-spacing": "off"
+    },
+    !noDeprecated && {
```
Comment (lydell) [inline index.js]:
```suggestion
    includeDeprecated && {
```

File: react.js [hunk 6]
```diff
@@ -1,21 +1,29 @@
 "use strict";
 
+const noDeprecated = process.env.ESLINT_CONFIG_PRETTIER_NO_DEPRECATED;
```
Comment (lydell) [inline react.js]:
```suggestion
const includeDeprecated = !process.env.ESLINT_CONFIG_PRETTIER_NO_DEPRECATED;
```

File: react.js [hunk 7]
```diff
@@ -1,21 +1,29 @@
...
+      "react/jsx-first-prop-new-line": "off",
+      "react/jsx-indent": "off",
+      "react/jsx-indent-props": "off",
+      "react/jsx-max-props-per-line": "off",
+      "react/jsx-one-expression-per-line": "off",
+      "react/jsx-props-no-multi-spaces": "off",
+      "react/jsx-tag-spacing": "off",
+      "react/jsx-wrap-multilines": "off"
+    },
+    !noDeprecated && {
```
Comment (lydell) [inline react.js]:
```suggestion
    includeDeprecated && {
```

File: index.js (hunk 5 above)
Comment (lydell) [reply index.js]: I thought the double negation was confusing. What do you think?

**Comment by alexilyaev** [comment]
@lydell I think I'm done.
The test was a bit tricky, see comments in my code.

**Comment by lydell** [comment]
Thanks! I’ll get back to you this weekend.

**Comment by alexilyaev** [comment]
@lydell Cool, I like the the updates you've made.

**Pull request merged**